src/cli/commands/test/mod.rs

```diff
@@ -0,0 +1,164 @@
+#[cfg(test)]
+mod tests;
+
+use regex::Regex;
+
+use std::fs;
+use serde_json::Value;
+use std::{collections::HashMap, fmt::Display, io::Write, path::PathBuf, str::from_utf8};
+use cairo_rs::{
+	cairo_run::cairo_run,
+	hint_processor::{
+		builtin_hint_processor::{
+			builtin_hint_processor_definition::{BuiltinHintProcessor, HintFunc},
+			hint_utils::get_integer_from_var_name,
+		},
+		hint_processor_definition::HintReference,
+		proxies::{exec_scopes_proxy::ExecutionScopesProxy, vm_proxy::VMProxy},
+	},
+	serde::deserialize_program::ApTracking,
+	vm::{errors::{vm_errors::VirtualMachineError}},
+};
+use clap::{Args, ValueHint};
+use log::error;
+use serde::Serialize;
+
+use super::CommandExecution;
+use super::list::{path_is_valid_directory, ListArgs};
+
+
+use crate::cli::commands::execute::compile::compile;
+
+#[derive(Args, Debug)]
+pub struct TestArgs {
+	/// Path to a cairo directory
+	#[clap(short, long, value_hint=ValueHint::DirPath, value_parser=path_is_valid_directory)]
+	pub root: PathBuf,
+
+
+
+
+}
+
+fn list_test_entrypoints(compiled_path: &PathBuf) -> Result<Vec<String>, String> {
+	let re = Regex::new(r"__main__.(test_\w+)").map_err(|_op| format!("Regex failed"))?;
+	let data = fs::read_to_string(compiled_path).map_err(|_op| format!("File does not exist"))?;
+	let json = serde_json::from_str::<Value>(&data).map_err(|_op|format!("Compile fails: {}", compiled_path.display()))?;
+	let mut test_entrypoints = Vec::new();
+	// if json is empty, throw error
+	if json.is_null() {
```

Comment:
This check is not needed

Reply:
actually this check is a hackish method to catch empty json, so unwrap would not be called at all

```
for (key, value) in json["identifiers"].as_object().unwrap() {
		if re.is_match(key) && value["type"] == "function" {	
			for capture in re.captures_iter(key) {
				test_entrypoints.push(capture[1].to_string());
			}
		}
	}
```

but i'm currently trying to find a better way to handle this